src/sync/sync-utils.ts

```diff
@@ -49,6 +54,57 @@ export async function findOrStartSync(
 	}, 0, logger);
 }
 
+type subscriptionUpdateTasks = {
+	totalNumberOfRepos?: number | null;
+	repositoryCursor?: string | null;
+	repositoryStatus?: string | null;
+}
+
+const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncType, targetTasks?: TaskType[]): Promise<void> => {
+	if (!syncType || !targetTasks?.length) {
+		return;
+	}
+
+	const subscriptionTasks = targetTasks.find(task => task === "repository");
```

Comment:
don't you mean `filter`?  or I guess you could change it to `hasRepositoryTask = targetTasks.includes("repository");`